mirrord/agent/src/entrypoint.rs

```diff
@@ -488,10 +489,19 @@ impl ClientConnectionHandler {
 }
 
 /// Initializes the agent's [`State`], channels, threads, and runs [`ClientConnectionHandler`]s.
-#[tracing::instrument(level = "trace", ret)]
-async fn start_agent(args: Args) -> Result<()> {
+#[tracing::instrument(level = Level::TRACE, ret, err)]
+async fn start_agent(args: Args) -> AgentResult<()> {
     trace!("start_agent -> Starting agent with args: {args:?}");
 
+    if let Some(metrics_address) = args.metrics.as_ref() {
+        let address = metrics_address.parse()?;
+        tokio::spawn(async move {
+            start_metrics(address)
+                .await
+                .inspect_err(|fail| tracing::error!(?fail, "Failed starting metrics server!"))
+        });
```

Comment:
Something should watch over the metrics task. Maybe the `CancellationToken` we have in `start_agent`?

Reply:
Help, how do I make this happen? Just `clone` the cancellation token before `tokio::spawn`?

Reply:
Yep

Reply:
Added the `cancellation_token` to axum.